consailapi/consultations/api/views.py

```diff
@@ -195,3 +195,39 @@ def create_reservation(
             ReservationSerializer(reservation).data,
             status=status.HTTP_200_OK,
         )
+
+
+class ReservationViewSet(
+    GenericViewSet,
+    ListModelMixin,
+    RetrieveModelMixin,
+    DestroyModelMixin,
+):
+    queryset = Reservation.objects.all().select_related("teacher", "student")
+    lookup_field = "uuid"
+    lookup_url_kwarg = "uuid"
+    permission_classes = (IsAuthenticated, IsStudentPermission)
+    serializer_class = ReservationSerializer
+
+    def get_queryset(self):
+        if "make_absent" in self.action:
+            return self.queryset.filter(teacher=self.request.user.teacher)  # noqa
+        return self.queryset.filter(student=self.request.user.student)  # noqa
+
+    def destroy(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        obj: Reservation = self.get_object()
+        obj.is_cancelled = True
+        obj.save()
```

Comment:
I think that we should also free the reservation slots so that other people can sign up for that consultation at this time